sdk/core/core-auth/.eslintrc.json

```diff
@@ -1,7 +1,7 @@
 {
   "plugins": ["@azure/azure-sdk"],
-  "extends": ["../../.eslintrc.old.json", "plugin:@azure/azure-sdk/recommended"],
-  "parserOptions": {
-    "createDefaultProgram": true
+  "extends": ["plugin:@azure/azure-sdk/azure-sdk-base"],
+  "rules": {
+    "@typescript-eslint/no-unused-vars": "off"
```

Comment:
Are we turning off the `no-unused-vars` rule in general or is this specific to this package?

Reply:
Only for this package, as far as I'm aware.

Reply:
yeah if you want to make a rule change global you can put it in the base config that this extends

Reply:
Perhaps to clarify my position:
This rule is only meant to reduce the issues for this specific project. We can remove this line and keep the issues, since eslint is still disabled. I wouldn't recommend disabling this rule for all of our projects.

Reply:
there is a better solution:

```
no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]
```

from https://eslint.org/docs/rules/no-unused-vars

Reply:
The only difference I see in the final report if we don't turn off this rule is an additional warning. There are 4 errors on empty function though

<img width="1031" alt="Screen Shot 2020-05-04 at 3 01 30 PM" src="https://user-images.githubusercontent.com/16890566/81017920-3eab6280-8e18-11ea-89cb-ef42951334d7.png">

Reply:
Fixed with https://github.com/Azure/azure-sdk-for-js/pull/8679/commits/bd1f96be013095be544f14814e5ebe7355aece72

Reply:
@sadasant Do we have an issue logged to handle the errors listed in https://gist.github.com/sadasant/3d39e8dbd5d33d9fd0ed95d7b8889c90?

Reply:
hey! Thanks for the commit. I'll make an issue for the missing lint issues.

Reply:
So, that commit broke the tests, but it made me realize how both fix this without breaking eslint, and how to fix the remaining eslint issues. This PR should go green now, with ESLint enabled! No need to make a new issue.

Reply:
@xirzec Oh! I also realized that `no-unused-vars` is set to `off` by default, and `@typescript-eslint/no-unused-vars` is only set to `warn`, so I could have let that slide. I will let it slide in this PR: https://github.com/Azure/azure-sdk-for-js/pull/8678

Reply:
Ah! That's interesting, the test result is different based on whether there is input or not...